progress/progress.go

```diff
@@ -80,23 +84,23 @@ func (p *Progress) Add(key string, state State) {
 func (p *Progress) render() {
 	p.mu.Lock()
 	defer p.mu.Unlock()
+	defer p.bw.Flush()
 
-	fmt.Fprint(p.w, "\033[?25l")
-	defer fmt.Fprint(p.w, "\033[?25h")
+	fmt.Fprint(p.bw, "\033[?2026h")
+	defer fmt.Fprint(p.bw, "\033[?2026l")
 
 	// clear already rendered progress lines
-	for i := range p.pos {
-		if i > 0 {
-			fmt.Fprint(p.w, "\033[A")
-		}
-		fmt.Fprint(p.w, "\033[2K\033[1G")
+	for range p.pos - 1 {
```

Comment:
This looks potentially dangerous for out-of-bounds errors, we should add a bounds check

Reply:
dope